In [1]:
import pandas as pd
import numpy as np

pd.set_option('display.max_columns', None)

In [2]:
df_data = pd.read_csv("1970-2021_natural_disasters-emdat_data.csv")
data = df_data

In [3]:
data.columns

Index(['Dis No', 'Year', 'Seq', 'Glide', 'Disaster Group', 'Disaster Subgroup',
       'Disaster Type', 'Disaster Subtype', 'Disaster Subsubtype',
       'Event Name', 'Country', 'ISO', 'Region', 'Continent', 'Location',
       'Origin', 'Associated Dis', 'Associated Dis2', 'OFDA Response',
       'Appeal', 'Declaration', 'Aid Contribution', 'Dis Mag Value',
       'Dis Mag Scale', 'Latitude', 'Longitude', 'Local Time', 'River Basin',
       'Start Year', 'Start Month', 'Start Day', 'End Year', 'End Month',
       'End Day', 'Total Deaths', 'No Injured', 'No Affected', 'No Homeless',
       'Total Affected', 'Reconstruction Costs ('000 US$)',
       'Insured Damages ('000 US$)', 'Total Damages ('000 US$)', 'CPI',
       'Adm Level', 'Admin1 Code', 'Admin2 Code', 'Geo Locations'],
      dtype='object')

In [4]:
columnas_deseadas = [
    'Dis No','Year', 'Seq', 'Glide', 'Disaster Group', 'Disaster Subgroup',
       'Disaster Type', 'Disaster Subtype', 'Disaster Subsubtype',
       'Event Name', 'Country', 'ISO', 'Region', 'Continent', 'Location',
       'Origin', 'Associated Dis', 'Associated Dis2', 'OFDA Response',
       'Appeal', 'Declaration', 'Aid Contribution'
]

df_miscol = df_data[columnas_deseadas]

In [5]:
df_miscol

Dis No  Year  Seq           Glide Disaster Group  \
0      1970-0013-ARG  1970   13             NaN        Natural   
1      1970-0109-AUS  1970  109             NaN        Natural   
2      1970-0044-BEN  1970   44             NaN        Natural   
3      1970-0063-BGD  1970   63             NaN        Natural   
4      1970-0026-BGD  1970   26             NaN        Natural   
...              ...   ...  ...             ...            ...   
14639  2021-0449-YEM  2021  449  FL-2021-000110        Natural   
14640  2021-0075-ZAF  2021   75             NaN        Natural   
14641  2021-0599-COD  2021  599  EP-2021-000138        Natural   
14642  2021-0020-SRB  2021   20             NaN        Natural   
14643  2021-0481-SSD  2021  481             NaN        Natural   

      Disaster Subgroup Disaster Type  Disaster Subtype Disaster Subsubtype  \
0          Hydrological         Flood               NaN                 NaN   
1        Meteorological         Storm  Tropical cyclone                 NaN   
2          Hydrological         Flood               NaN                 NaN   
3        Meteorological         Storm  Tropical cyclone                 NaN   
4        Meteorological         Storm               NaN                 NaN   
...                 ...           ...               ...                 ...   
14639      Hydrological         Flood               NaN                 NaN   
14640      Hydrological         Flood               NaN                 NaN   
14641        Biological      Epidemic     Viral disease                 NaN   
14642      Hydrological         Flood               NaN                 NaN   
14643      Hydrological         Flood               NaN                 NaN   

       Event Name                                 Country  ISO  \
0             NaN                               Argentina  ARG   
1             Ada                               Australia  AUS   
2             NaN                                   Benin  BEN   
3             NaN                              Bangladesh  BGD   
4             NaN                              Bangladesh  BGD   
...           ...                                     ...  ...   
14639         NaN                                   Yemen  YEM   
14640         NaN                            South Africa  ZAF   
14641  Meningitis  Congo (the Democratic Republic of the)  COD   
14642         NaN                                  Serbia  SRB   
14643         NaN                             South Sudan  SSD   

                          Region Continent  \
0                  South America  Americas   
1      Australia and New Zealand   Oceania   
2                 Western Africa    Africa   
3                  Southern Asia      Asia   
4                  Southern Asia      Asia   
...                          ...       ...   
14639               Western Asia      Asia   
14640            Southern Africa    Africa   
14641              Middle Africa    Africa   
14642            Southern Europe    Europe   
14643            Northern Africa    Africa   

                                                Location       Origin  \
0                                                Mendoza          NaN   
1                                             Queensland          NaN   
2                                         Atacora region          NaN   
3                                     Khulna, Chittagong          NaN   
4                                                    NaN          NaN   
...                                                  ...          ...   
14639  Dhamar, Amran, Al Mahwit, Marib, Ibb, Sana’a C...  Heavy rains   
14640  Mpumalanga Province, Free State Province and t...  Heavy rains   
14641                                    Tshopo province          NaN   
14642  Zitoradja (Toplicki), Doljevac (Nisavski) , Di...  Heavy rains   
14643  Mayendit County (Unity); Ayod, Fangak Counties...  Heavy rains   

      Associated Dis Associated Dis2 OFDA Respons

In [6]:
df_miscol.isnull().sum()

Dis No                     0
Year                       0
Seq                        0
Glide                  13063
Disaster Group             0
Disaster Subgroup          0
Disaster Type              0
Disaster Subtype        2747
Disaster Subsubtype    13600
Event Name             10999
Country                    0
ISO                        0
Region                     0
Continent                  0
Location                1346
Origin                 10864
Associated Dis         11412
Associated Dis2        13946
OFDA Response          13194
Appeal                 12204
Declaration            11517
Aid Contribution       13967
dtype: int64

In [7]:
df_miscol.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14644 entries, 0 to 14643
Data columns (total 22 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Dis No               14644 non-null  object 
 1   Year                 14644 non-null  int64  
 2   Seq                  14644 non-null  int64  
 3   Glide                1581 non-null   object 
 4   Disaster Group       14644 non-null  object 
 5   Disaster Subgroup    14644 non-null  object 
 6   Disaster Type        14644 non-null  object 
 7   Disaster Subtype     11897 non-null  object 
 8   Disaster Subsubtype  1044 non-null   object 
 9   Event Name           3645 non-null   object 
 10  Country              14644 non-null  object 
 11  ISO                  14644 non-null  object 
 12  Region               14644 non-null  object 
 13  Continent            14644 non-null  object 
 14  Location             13298 non-null  object 
 15  Origin               3780 non-null  

-----------------

#### COLUMNA Dis No: Código único de desastre

- Índice 1
- Datos de tipo objeto (string)
- Incluye el año, Seq y ISO

In [8]:
df_data['Dis No'].value_counts()

Dis No
1970-0013-ARG    1
2008-0506-SOM    1
2008-0257-SUR    1
2008-9755-SYR    1
2008-0384-TCA    1
                ..
1996-0468-SDN    1
1996-0126-SDN    1
1996-9220-SDN    1
1996-0130-SLE    1
2021-0481-SSD    1
Name: count, Length: 14644, dtype: int64

#### COLUMNA Year: Año del desastre

- Índice 2
- Pensamos dejarla si conseguimos calcular la Duration para facilitar visualizaciones 

In [9]:
df_data['Year'].value_counts()

Year
2002    532
2000    523
2005    498
2007    449
2001    447
2006    446
2019    442
2010    441
1999    416
2004    405
2008    400
2015    398
2020    398
2003    392
2009    384
2017    371
2012    370
1998    363
2011    357
2013    353
2016    350
2014    348
2018    338
1997    323
2021    315
1990    303
1995    277
1996    273
1993    267
1991    266
1994    255
1988    234
1992    232
1987    227
1983    206
1989    189
1985    175
1986    174
1984    156
1982    150
1981    146
1980    144
1977    141
1978    137
1979    122
1976     99
1970     82
1974     72
1975     67
1973     65
1972     63
1971     63
Name: count, dtype: int64

#### COLUMNA Seq – Número de secuencia en el año según la base de datos EM-DAT.

- Índice 3
- La eliminamos ya que no aporta información extra al Dis No

#### COLUMNA Glide: Global Unique Disaster Identifier

- Índice 4
- 89% de nulos y datos sin valor. La eliminamos.

#### COLUMNA Disaster Group – Grupo principal (ej. Natural, Technological)

- Índice 5
- Datos de tipo objeto (string)
- No hay valores nulos, pero todos los valores son "Natural". Eliminamos la columna.

In [10]:
df_data['Disaster Group'].value_counts()

Disaster Group
Natural    14644
Name: count, dtype: int64

In [11]:
df_data['Disaster Group'].unique()

array(['Natural'], dtype=object)

#### COLUMNA Disaster Subgroup – Subgrupo (ej. Meteorological, Hydrological)

- Índice 6
- Datos de tipo objeto (string)
- No hay valores nulos, pero sí observamos categorías del desastre.

In [12]:
df_data['Disaster Subgroup'].value_counts()

Disaster Subgroup
Hydrological         5985
Meteorological       4589
Biological           1530
Geophysical          1400
Climatological       1139
Extra-terrestrial       1
Name: count, dtype: int64

#### COLUMNA Disaster Type – Tipo (ej. Storm, Flood)

- Índice 7
- Datos de tipo objeto (string)
- No hay valores nulos, encontramos valores concretos sobre el tipo de desastre

In [13]:
df_data['Disaster Type'].value_counts()

Disaster Type
Flood                    5272
Storm                    4005
Epidemic                 1436
Earthquake               1137
Landslide                 713
Drought                   685
Extreme temperature       584
Wildfire                  452
Volcanic activity         222
Insect infestation         93
Mass movement (dry)        41
Glacial lake outburst       2
Impact                      1
Animal accident             1
Name: count, dtype: int64

In [14]:
df_data['Disaster Type'].unique()

array(['Flood', 'Storm', 'Drought', 'Landslide', 'Earthquake', 'Epidemic',
       'Wildfire', 'Mass movement (dry)', 'Extreme temperature ',
       'Volcanic activity', 'Insect infestation', 'Impact',
       'Animal accident', 'Glacial lake outburst'], dtype=object)

#### COLUMNA Disaster Subtype – Subtipo (ej. Tropical cyclone)

- Índice 8
- Datos de tipo objeto (string)
- 2747 nulos - Repetimos información de "Disaster Type" para conservar los que sí tenemos info.

In [15]:
df_miscol["Disaster Subtype"] = df_miscol["Disaster Subtype"].fillna(df_miscol["Disaster Type"])

C:\Users\PC\AppData\Local\Temp\ipykernel_5248\1272324094.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_miscol["Disaster Subtype"] = df_miscol["Disaster Subtype"].fillna(df_miscol["Disaster Type"])


In [16]:
df_miscol['Disaster Subtype'].unique()

array(['Flood', 'Tropical cyclone', 'Storm', 'Drought', 'Avalanche',
       'Ground movement', 'Bacterial disease', 'Flash flood',
       'Land fire (Brush, Bush, Pasture)', 'Landslide', 'Riverine flood',
       'Convective storm', 'Cold wave', 'Heat wave', 'Ash fall',
       'Viral disease', 'Forest fire', 'Coastal flood',
       'Insect infestation', 'Tsunami', 'Parasitic disease', 'Locust',
       'Severe winter conditions', 'Rockfall', 'Epidemic', 'Grasshopper',
       'Mudslide', 'Subsidence', 'Extra-tropical storm', 'Wildfire',
       'Mass movement (dry)', 'Impact', 'Lava flow', 'Volcanic activity',
       'Animal accident', 'Pyroclastic flow ', 'Earthquake',
       'Glacial lake outburst'], dtype=object)

In [17]:
print("Nulos después:", df_miscol["Disaster Subtype"].isna().sum())

Nulos después: 0


In [18]:
df_miscol['Disaster Subtype'].value_counts()

Disaster Subtype
Riverine flood                      2633
Tropical cyclone                    2066
Flood                               1788
Ground movement                     1097
Convective storm                    1084
Flash flood                          770
Bacterial disease                    744
Storm                                724
Drought                              685
Landslide                            555
Viral disease                        509
Forest fire                          303
Cold wave                            302
Heat wave                            205
Ash fall                             203
Epidemic                             137
Extra-tropical storm                 131
Land fire (Brush, Bush, Pasture)     115
Avalanche                            111
Coastal flood                         81
Severe winter conditions              77
Mudslide                              75
Locust                                59
Parasitic disease                     46

#### COLUMNA Disaster Subsubtype – Sub-subtipo (si aplica)

- índice 9
- 93% de nulos, la eliminamos. Todos los subtypes tienen subsubtype, no nos ayuda para resolver los nulos de la columna anterior.

In [19]:
mask = df_data["Disaster Subtype"].isna() & df_data["Disaster Subsubtype"].notna()
print("Casos con Subsubtype pero sin Subtype:", mask.sum())

Casos con Subsubtype pero sin Subtype: 0


#### COLUMNA Event Name – Nombre del evento (si existe)

- Índice 10
- 75% de nulos - DE MOMENTO DEJAMOS como NAN

- Todos los Disaster Type tienen al menos un nulo en Event Name

* La mayoría provienen de "Flood" - 5261, siguiendo "Storm" - 1840 y "Earthquake" - 1136
* No la eliminamos, de las 3645 filas que tenemos nombre posteriormente podemos sacar info/gráficas.  

In [20]:
df_miscol['Event Name'].isnull().sum()

10999

In [21]:
nulos_por_type = (
    df_miscol.groupby("Disaster Type")["Event Name"]
      .apply(lambda x: x.isna().sum())
      .reset_index(name="Nombre nulo")
)

nulos_por_type


Disaster Type  Nombre nulo
0         Animal accident            1
1                 Drought          685
2              Earthquake         1136
3                Epidemic          268
4    Extreme temperature           583
5                   Flood         5261
6   Glacial lake outburst            2
7                  Impact            1
8      Insect infestation           36
9               Landslide          711
10    Mass movement (dry)           40
11                  Storm         1840
12      Volcanic activity            7
13               Wildfire          428

In [22]:
df_sin_nulos = df_miscol[df_miscol['Event Name'].notna()][['Disaster Type', 'Event Name']]
df_sin_nulos

Disaster Type                          Event Name
1                  Storm                                 Ada
14                 Storm                                Fran
15              Epidemic                             Cholera
23              Epidemic                             Cholera
28              Epidemic                             Cholera
...                  ...                                 ...
14629           Wildfire                       Bootleg Fire 
14632  Volcanic activity                       La Souffrière
14633              Storm                    Hurricane 'Elsa'
14637              Storm  Tropical cyclone 'Conson' (Jolina)
14641           Epidemic                          Meningitis

[3645 rows x 2 columns]

#### COLUMNA Country – País afectado

- Índice 11
- Datos de tipo objeto (string)
- No disponemos de valores nulos.
- El país que encabeza nuestra lista es USA, seguido de China.
- Filtrado realizado para evitar reiteraciones y algunos nombres antiguos resueltos (Checoslovaquia - Rep Checa). URSS conservada ya que no abarcaba sólo Rusia.

In [23]:
sorted(df_miscol["Country"].unique())

['Afghanistan',
 'Albania',
 'Algeria',
 'American Samoa',
 'Angola',
 'Anguilla',
 'Antigua and Barbuda',
 'Argentina',
 'Armenia',
 'Australia',
 'Austria',
 'Azerbaijan',
 'Azores Islands',
 'Bahamas (the)',
 'Bahrain',
 'Bangladesh',
 'Barbados',
 'Belarus',
 'Belgium',
 'Belize',
 'Benin',
 'Bermuda',
 'Bhutan',
 'Bolivia (Plurinational State of)',
 'Bosnia and Herzegovina',
 'Botswana',
 'Brazil',
 'Brunei Darussalam',
 'Bulgaria',
 'Burkina Faso',
 'Burundi',
 'Cabo Verde',
 'Cambodia',
 'Cameroon',
 'Canada',
 'Canary Is',
 'Cayman Islands (the)',
 'Central African Republic',
 'Chad',
 'Chile',
 'China',
 'Colombia',
 'Comoros (the)',
 'Congo (the Democratic Republic of the)',
 'Congo (the)',
 'Cook Islands (the)',
 'Costa Rica',
 'Croatia',
 'Cuba',
 'Cyprus',
 'Czech Republic (the)',
 'Czechoslovakia',
 'Côte d’Ivoire',
 'Denmark',
 'Djibouti',
 'Dominica',
 'Dominican Republic (the)',
 'Ecuador',
 'Egypt',
 'El Salvador',
 'Equatorial Guinea',
 'Eritrea',
 'Estonia',
 'Ethio

In [24]:
# Diccionario de normalización de países
replace_dict = {
    # Congo
    "Congo (the Democratic Republic of the)": "Democratic Republic of the Congo",
    "Congo (the)": "Republic of the Congo",

    # Czechia / Checoslovaquia
    "Czech Republic (the)": "Czech Republic",
    "Czechoslovakia": "Czech Republic",  # ya no existe, se fusiona con Czechia

    # Alemania
    "Germany Fed Rep": "Germany",
    "Germany Dem Rep": "Germany",

    # Guinea
    "Guinea-Bissau": "Guinea-Bissau",
    "Guinea": "Guinea",

    # Yemen
    "Yemen Arab Rep": "Yemen",
    "Yemen P Dem Rep": "Yemen",
}

# Aplicar reemplazo
df_miscol["Country"] = df_miscol["Country"].replace(replace_dict)

C:\Users\PC\AppData\Local\Temp\ipykernel_5248\2851945740.py:25: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_miscol["Country"] = df_miscol["Country"].replace(replace_dict)


In [25]:
df_miscol["Country"].value_counts().head(20)

Country
United States of America (the)    949
China                             906
India                             658
Philippines (the)                 612
Indonesia                         528
Bangladesh                        317
Japan                             272
Mexico                            262
Viet Nam                          241
Australia                         240
Brazil                            227
Pakistan                          213
Iran (Islamic Republic of)        205
Afghanistan                       200
Colombia                          188
Peru                              178
France                            173
Russian Federation (the)          173
Thailand                          155
Italy                             147
Name: count, dtype: int64

#### COLUMNA ISO – Código ISO del país

- Índice 12
- Ya que aparece incluído en 'Dis No' y podemos averiguarlo, eliminamos.

In [26]:
df_miscol=df_miscol.drop(columns=['Seq', 'Glide', 'Disaster Group', 'Disaster Subsubtype', 'ISO'])

#### COLUMNA Region – Región (ej. Americas, Asia)

- Índice 13
- Datos de tipo objeto (string)
- No disponemos de valores nulos. Completa la información de "Continente". Dejamos ambas para facilitar las visualizaciones (por si queremos filtrar por Continente o región posteriormente).

In [27]:
df_miscol[['Region', 'Continent']].head()

Region Continent
0              South America  Americas
1  Australia and New Zealand   Oceania
2             Western Africa    Africa
3              Southern Asia      Asia
4              Southern Asia      Asia

In [28]:
# Tabla de frecuencia cruzada
tabla_region_cont = pd.crosstab(df_miscol['Region'], df_miscol['Continent'])

tabla_region_cont

Continent                  Africa  Americas  Asia  Europe  Oceania
Region                                                            
Australia and New Zealand       0         0     0       0      312
Caribbean                       0       530     0       0        0
Central America                 0       751     0       0        0
Central Asia                    0         0   141       0        0
Eastern Africa               1128         0     0       0        0
Eastern Asia                    0         0  1578       0        0
Eastern Europe                  0         0     0     535        0
Melanesia                       0         0     0       0      232
Micronesia                      0         0     0       0       43
Middle Africa                 421         0     0       0        0
Northern Africa               305         0     0       0        0
Northern America                0      1075     0       0        0
Northern Europe                 0         0     0     207        0
Polynesia                       0         0     0       0       75
Russian Federation              0         0     0      47        0
South America                   0      1163     0       0        0
South-Eastern Asia              0         0  1809       0        0
Southern Africa               200         0     0       0        0
Southern Asia                   0         0  1849       0        0
Southern Europe                 0         0     0     577        0
Western Africa                758         0     0       0        0
Western Asia                    0         0   416       0        0
Western Europe                  0         0     0     492        0

#### COLUMNA Continent - Continente

- Índice 14
- Datos de tipo objeto (string)
- No tenemos valores nulos

#### COLUMNA Location – Lugar específico

- Índice 15
- 1346 nulos. Los modifico reiterando la información de "Region", para no perder información pero eliminar los nulos (similar a lo que hice con Types y Subtypes)

In [29]:
df_miscol["Location"] = df_miscol["Location"].fillna(df_miscol["Region"])

In [30]:
print("Nulos después:", df_miscol["Location"].isna().sum())


Nulos después: 0


In [31]:
df_miscol.sample(5)

Dis No  Year Disaster Subgroup Disaster Type   Disaster Subtype  \
4892   1996-0490-SOM  1996        Biological      Epidemic  Bacterial disease   
12105  2014-0271-JPN  2014      Hydrological     Landslide          Landslide   
2802   1993-0462-CHN  1993    Meteorological         Storm              Storm   
13784  2020-0060-ARG  2020    Meteorological         Storm   Convective storm   
4992   1997-0280-UGA  1997      Hydrological     Landslide          Landslide   

      Event Name    Country          Region Continent  \
4892     Cholera    Somalia  Eastern Africa    Africa   
12105        NaN      Japan    Eastern Asia      Asia   
2802       Levis      China    Eastern Asia      Asia   
13784        NaN  Argentina   South America  Americas   
4992         NaN     Uganda  Eastern Africa    Africa   

                                                Location               Origin  \
4892   Mogadishu, Merca, Middle Shabelle and Galgaduu...                  NaN   
12105  Hirosimasi_Asaminamiku, Hirosimasi_Asakitaku d...  Torrential rainfall   
2802                                     Hainan province                  NaN   
13784  Famaillá, La Reducción, La Madrid, Los Ralos, ...                  NaN   
4992                                        Mbale Region                  NaN   

      Associated Dis Associated Dis2 OFDA Response Appeal Declaration  \
4892             NaN             NaN           NaN     No          No   
12105            NaN             NaN           NaN    NaN         NaN   
2802             NaN             NaN           NaN    NaN         NaN   
13784          Flood             NaN           NaN    NaN         NaN   
4992             NaN             NaN           NaN    NaN         NaN   

       Aid Contribution  
4892                NaN  
12105               NaN  
2802                NaN  
13784               NaN  
4992                NaN

-----------------

#### COLUMNA Origin – Origen del desastre (ej. Natural, Industrial)

- Índice 16
- 10864 nulos, he intentado agrupar por categorías pero es imposible ya que agrupa muchos en "Other" debido a typos y textos largos. De momento la dejo por si queremos revisar algún dato concreto.

In [32]:
df_miscol["Origin"].isnull().sum()

10864

In [33]:
df_miscol["Origin"].unique()

array([nan, 'Typhoon ', 'Heavy rains', 'Landslide', 'Heavy rain',
       'Earthquake and landslide', 'Famine', 'El Nino',
       'Extra-tropical cyclone', 'Brief torrential rain',
       'Heavy rains and thunderstorms', 'Monsoonal rain',
       'Monssonal rain', 'Heavy rains, tremor', 'Brief torrential rains',
       'Monsoon rain', 'Tropical cyclone', 'Drought period',
       'Prolonged drought', 'Monsoon rains',
       'Volcanic activity, tropical rainstorms and ash from the eruptions',
       'monsoonal rain', 'Heavy rain and Snowmelt',
       'Snowmelt and Heavy rains', 'Rain, snowmelt', 'Earthquake',
       'Snowmelt', 'Hevay rain', 'Typhoon Gali', 'Mosoonal rain',
       'Heavy rain and brief torrential rain',
       'Breakdown water supply, poor hygiene practices and increase of number of displaced perdons',
       'Flood', 'Heavy rains, melting snow', 'Continous rainfall',
       'Snow storm and heavy winds', 'Rain and snowmelt', 'Lack rainfall',
       'Food consumed', 'El Nin

#### COLUMNA Associated Dis – Desastres asociados (1)

- Índice 17
- 78% de nulos, intento de rellenar con la info de Associated Dis2 pero tenemos 95% nulos en esa. Aun así, nos falta mucha info (11403). 

*Eliminar? 

In [34]:
df_data["Associated Dis"].isnull().sum()

11412

#### COLUMNA Associated Dis2 – Desastres asociados (2)

- Índice 18
- 95% de nulos. Tras intentar sacar info para completar la anterior, eliminamos.

In [35]:
df_miscol["Associated Dis"] = df_miscol["Associated Dis"].fillna(df_miscol["Associated Dis2"])

In [36]:
df_miscol["Associated Dis"].unique()


array([nan, 'Tsunami/Tidal wave', 'Famine', 'Broken Dam/Burst bank',
       'Food shortage', 'Slide (land, mud, snow, rock)', 'Wildfire',
       'Hail', 'Heat wave', 'Crop failure', 'Surge', 'Storm', 'Flood',
       'Water shortage', 'Snow/ice', 'Epidemic', 'Cold wave', 'Drought',
       'Rain', 'Volcanic activity', 'Pollution',
       'Avalanche (Snow, Debris)', 'Transport accident', 'Fire',
       'Insect infestation', 'Collapse', 'Earthquake', 'Explosion',
       'Lightening', 'Fog', 'Industrial accidents'], dtype=object)

In [37]:
df_miscol["Associated Dis"].isnull().sum()

11403

#### COLUMNA OFDA Response – Respuesta de la Office of U.S. Foreign Disaster Assistance

- Índice 19
- 13194 nulos. Eliminamos. 

In [38]:
df_miscol["OFDA Response"].isnull().sum()

13194

In [39]:
df_miscol.drop(columns=["OFDA Response", ])

Dis No  Year Disaster Subgroup Disaster Type  Disaster Subtype  \
0      1970-0013-ARG  1970      Hydrological         Flood             Flood   
1      1970-0109-AUS  1970    Meteorological         Storm  Tropical cyclone   
2      1970-0044-BEN  1970      Hydrological         Flood             Flood   
3      1970-0063-BGD  1970    Meteorological         Storm  Tropical cyclone   
4      1970-0026-BGD  1970    Meteorological         Storm             Storm   
...              ...   ...               ...           ...               ...   
14639  2021-0449-YEM  2021      Hydrological         Flood             Flood   
14640  2021-0075-ZAF  2021      Hydrological         Flood             Flood   
14641  2021-0599-COD  2021        Biological      Epidemic     Viral disease   
14642  2021-0020-SRB  2021      Hydrological         Flood             Flood   
14643  2021-0481-SSD  2021      Hydrological         Flood             Flood   

       Event Name                           Country  \
0             NaN                         Argentina   
1             Ada                         Australia   
2             NaN                             Benin   
3             NaN                        Bangladesh   
4             NaN                        Bangladesh   
...           ...                               ...   
14639         NaN                             Yemen   
14640         NaN                      South Africa   
14641  Meningitis  Democratic Republic of the Congo   
14642         NaN                            Serbia   
14643         NaN                       South Sudan   

                          Region Continent  \
0                  South America  Americas   
1      Australia and New Zealand   Oceania   
2                 Western Africa    Africa   
3                  Southern Asia      Asia   
4                  Southern Asia      Asia   
...                          ...       ...   
14639               Western Asia      Asia   
14640            Southern Africa    Africa   
14641              Middle Africa    Africa   
14642            Southern Europe    Europe   
14643            Northern Africa    Africa   

                                                Location       Origin  \
0                                                Mendoza          NaN   
1                                             Queensland          NaN   
2                                         Atacora region          NaN   
3                                     Khulna, Chittagong          NaN   
4                                          Southern Asia          NaN   
...                                                  ...          ...   
14639  Dhamar, Amran, Al Mahwit, Marib, Ibb, Sana’a C...  Heavy rains   
14640  Mpumalanga Province, Free State Province and t...  Heavy rains   
14641                                    Tshopo province          NaN   
14642  Zitoradja (Toplicki), Doljevac (Nisavski) , Di...  Heavy rains   
14643  Mayendit County (Unity); Ayod, Fangak Counties...  Heavy rains   

      Associated Dis Associated Dis2 Appeal Declaration  Aid Contribution  
0                NaN             NaN    NaN         NaN               NaN  
1                NaN             NaN    NaN         NaN               NaN  
2                NaN             NaN    NaN         NaN               NaN  
3                NaN             NaN    NaN         NaN               NaN  
4                NaN             NaN    NaN         NaN               NaN  
...              ...             ...    ...         ...               ...  
14639            NaN             NaN    NaN         NaN               NaN  
14640            NaN             NaN    NaN         NaN               NaN  
14641            NaN             NaN    NaN         Yes               NaN  
14642            NaN             NaN    NaN         Yes               NaN  
14643            NaN             NaN    NaN         NaN               NaN  

[14644 rows x 16 columns]

#### COLUMNA Appeal – Si hubo apelación internacional

- Índice 20
- 12204 nulos. Considerar eliminar ?

In [40]:
df_miscol["Appeal"].isnull().sum()

12204

#### COLUMNA Declaration – Declaración oficial de desastre.

- Índice 21
- 11517 nulos. Podemos considerar que la mayoría no se consideraron oficialmente, o no disponemos de registros.

In [41]:
df_miscol["Declaration"].isnull().sum()

11517

#### COLUMNA Aid Contribution – Contribución de ayuda (si se registró)

- Índice 22
- 95% de nulos y muchos sin registrar. Eliminamos

In [42]:
df_miscol["Aid Contribution"].isnull().sum()

13967

In [44]:
df_miscol.drop(columns=["Associated Dis2","OFDA Response", "Aid Contribution","Appeal","Declaration"])

Dis No  Year Disaster Subgroup Disaster Type  Disaster Subtype  \
0      1970-0013-ARG  1970      Hydrological         Flood             Flood   
1      1970-0109-AUS  1970    Meteorological         Storm  Tropical cyclone   
2      1970-0044-BEN  1970      Hydrological         Flood             Flood   
3      1970-0063-BGD  1970    Meteorological         Storm  Tropical cyclone   
4      1970-0026-BGD  1970    Meteorological         Storm             Storm   
...              ...   ...               ...           ...               ...   
14639  2021-0449-YEM  2021      Hydrological         Flood             Flood   
14640  2021-0075-ZAF  2021      Hydrological         Flood             Flood   
14641  2021-0599-COD  2021        Biological      Epidemic     Viral disease   
14642  2021-0020-SRB  2021      Hydrological         Flood             Flood   
14643  2021-0481-SSD  2021      Hydrological         Flood             Flood   

       Event Name                           Country  \
0             NaN                         Argentina   
1             Ada                         Australia   
2             NaN                             Benin   
3             NaN                        Bangladesh   
4             NaN                        Bangladesh   
...           ...                               ...   
14639         NaN                             Yemen   
14640         NaN                      South Africa   
14641  Meningitis  Democratic Republic of the Congo   
14642         NaN                            Serbia   
14643         NaN                       South Sudan   

                          Region Continent  \
0                  South America  Americas   
1      Australia and New Zealand   Oceania   
2                 Western Africa    Africa   
3                  Southern Asia      Asia   
4                  Southern Asia      Asia   
...                          ...       ...   
14639               Western Asia      Asia   
14640            Southern Africa    Africa   
14641              Middle Africa    Africa   
14642            Southern Europe    Europe   
14643            Northern Africa    Africa   

                                                Location       Origin  \
0                                                Mendoza          NaN   
1                                             Queensland          NaN   
2                                         Atacora region          NaN   
3                                     Khulna, Chittagong          NaN   
4                                          Southern Asia          NaN   
...                                                  ...          ...   
14639  Dhamar, Amran, Al Mahwit, Marib, Ibb, Sana’a C...  Heavy rains   
14640  Mpumalanga Province, Free State Province and t...  Heavy rains   
14641                                    Tshopo province          NaN   
14642  Zitoradja (Toplicki), Doljevac (Nisavski) , Di...  Heavy rains   
14643  Mayendit County (Unity); Ayod, Fangak Counties...  Heavy rains   

      Associated Dis  
0                NaN  
1                NaN  
2                NaN  
3                NaN  
4                NaN  
...              ...  
14639            NaN  
14640            NaN  
14641            NaN  
14642            NaN  
14643            NaN  

[14644 rows x 12 columns]